### PR TITLE
[CALCITE-6184] Add checkNullTest on SqlOperatorTest

### PR DESCRIPTION
**Pull request by caicancai**
I noticed that some examples of SqlOperatorTest lack checknull tests. I am adding them one by one, so this PR can not be merged yet.
https://issues.apache.org/jira/browse/CALCITE-6184

### CODE REVIEW

**Review by XuQianJin-Stars — APPROVED** [review]

**Comment by XuQianJin-Stars** [comment]
hi @caicancai When you have finished adding it, please ping me.

**Comment by caicancai** [comment]
@XuQianJin-Stars finish

**Comment by JiajunBernoulli** [comment]
Should we use `[MINOR]`?

Here is a discussion: https://lists.apache.org/thread/39mfnbz8sqborty1lqs1j2kpx656g5o4

**Comment by caicancai** [comment]
> Should we use `[MINOR]`?
> 
> Here is a discussion: https://lists.apache.org/thread/39mfnbz8sqborty1lqs1j2kpx656g5o4

Thank you. It seems that it is necessary to open a jira for my PR. I will sort out which tests should add checknull and which tests should not add checknull. At present, it seems that adding checknull is necessary for some.

**Comment by caicancai** [comment]
Please wait until my jira is completed before merging it. Thank you.

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 1]
```diff
@@ -12230,6 +12235,8 @@ void testTimestampDiff(boolean coercionEnabled) {
     f.checkScalar("time_trunc(time '12:34:56', hour)",
         "12:00:00", "TIME(0) NOT NULL");
     f.checkNull("time_trunc(cast(null as time), second)");
+    f.checkNull("time_trunc(cast(null as time), minute)");
```
Comment (tanclary) [inline testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:12238]: I don't really see how these tests help us. We already have a `checkNull` test for `second`, the value of the 2nd operand isn't going to influence the behavior (as long as it is the same type).

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 2]
```diff
@@ -12292,6 +12299,18 @@ void testTimestampDiff(boolean coercionEnabled) {
             + "'TIMESTAMP_TRUNC\\(<TIME\\(0\\)>, <INTERVAL HOUR>\\)'\\. "
             + "Supported form\\(s\\): 'TIMESTAMP_TRUNC\\(<TIMESTAMP>, <DATETIME_INTERVAL>\\)'",
         false);
+    f.checkNull("timestamp_trunc(CAST(NULL AS TIMESTAMP), second)");
```
Comment (tanclary) [inline testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:12302]: Same comment as above. Maybe one or two of these is good but an exhaustive list doesn't add much value other than making the test methods longer than necessary.

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java [hunk 3]
```diff
@@ -12348,6 +12367,18 @@ void testTimestampDiff(boolean coercionEnabled) {
             + "'DATETIME_TRUNC\\(<TIME\\(0\\)>, <INTERVAL HOUR>\\)'\\. "
             + "Supported form\\(s\\): 'DATETIME_TRUNC\\(<TIMESTAMP>, <DATETIME_INTERVAL>\\)'",
         false);
+    f.checkNull("datetime_trunc(CAST(NULL AS TIMESTAMP), second)");
```
Comment (tanclary) [inline testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:12370]: Are there tests for when the second operand is null? If not, that would be a worthwhile test to add. Same comment for other methods.

**Review by tanclary — CHANGES_REQUESTED** [review]
I think if we are going to add tests we should make sure they are testing different things (i.e. not redundant). Adding 2 tests that test 2 different things is more valuable than adding 10 tests that test 1 thing (in my own opinion).

File: testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java (hunk 1 above)
Comment (caicancai) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:12238]: I just tested and it seems that the second parameter is null is not allowed, which will fail to parse

Comment (caicancai) [reply testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java:12238]: thanks

**Comment by caicancai** [comment]
My fork repository seems to have exploded, I will mention a new PR

**Pull request closed without merging**